.github/ISSUE_TEMPLATE/release.md

```diff
@@ -1,59 +1,59 @@
 ---
-name: k6 Release 
-about: k6 release accommodates activities and a checklist that has the k6 open-source release process.
+name: k6 Release
+about: k6 release accommodates activities and a checklist with the k6 open-source release process.
 title: 'k6 release v0.4x.0'
 ---
 
-**Release Date**: 
+**Release Date**:
 
 25th May 2023 <<  TODO: WRITE HERE THE UPDATED RELEASE DATE
 
 ## Release Activities
 
 ### At the beginning of the cycle
 
-- [ ] Create a new branch `release-notes-v04x` and add a new related [release notes file](release-template/release%20notes) using the available [template](release-template/release%20notes/template.md).
-- [ ] Go through the potential [dependencies updates](Dependencies.md) then create a PR if any is relevant. 
+- [ ] Create a new `release-notes-v04x` branch and add a new [release notes file](release-template/release%20notes) using the available [template](release-template/release%20notes/template.md) to the [repository's `release notes` folder](https://github.com/grafana/k6/tree/cbd9e9ad184f19fdd84c7041f8084218ce4b738f/release%20notes).
```

Comment:
Maybe also have the branch be `release-notes-v0.4x.0` so that it matches the semver a lot closer ?